app/src/main/java/de/tum/in/tumcampusapp/component/other/generic/activity/ActivityForDownloadingExternal.java

```diff
@@ -39,8 +39,8 @@ public ActivityForDownloadingExternal(String method, int layoutId) {
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        LocalBroadcastManager.getInstance(this)
-                             .registerReceiver(receiver, new IntentFilter(DownloadService.BROADCAST_NAME));
+        IntentFilter filter = new IntentFilter(DownloadService.BROADCAST_NAME);
+        LocalBroadcastManager.getInstance(this).registerReceiver(receiver, filter);
```

Comment:
I think `registerReceiver()` should be done in `onStart()`.
Since we unregister it in `onStop()` we might end up with no receiver.

Reply:
Makes sense.